FaceDetectionServer/client/multi_client.py

```diff
@@ -0,0 +1,437 @@
+"""
+Client script that can use multiple communication methods to send images to the
+computervision backend and measure latency.
+"""
+import sys
+import os
+import platform
+import requests
+import subprocess
+import re
+import base64
+import socket
+import websocket
+import struct
+import json
+import time
+import logging
+from threading import Thread
+
+WEBSOCKET_OPCODE_BINARY = 0x2
+PING_INTERVAL = 4
+TEST_PASS = False
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(threadName)s - %(levelname)s - %(message)s')
+fh = logging.FileHandler('multi_client.log')
+fh.setLevel(logging.DEBUG)
+fh.setFormatter(formatter)
+logger.addHandler(fh)
+ch = logging.StreamHandler()
+ch.setLevel(logging.INFO)
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+if platform.system() == "Darwin":
+    PING = "/sbin/ping"
+    PING_REGEX = r'round-trip min/avg/max/stddev = (.*)/(.*)/(.*)/(.*) ms'
+else:
+    PING = "/bin/ping"
+    PING_REGEX = r'rtt min/avg/max/mdev = (.*)/(.*)/(.*)/(.*) ms'
+
+class Client:
```

Comment:
Do we need to track stddev as well?

Reply:
That would be a good addition. We couldn't use the value from "ping" itself since it is executed with only one sample at a time (-c 1). We'd have to calculate it ourselves, but we have plenty of sample code showing how to do that, so it would be a reasonably simple update.